wideq/core.py

```diff
@@ -240,166 +345,218 @@ def parse_oauth_callback(url):
-                        {'countryCode': country, 'langCode': language})
-        return cls(gw['empUri'], gw['thinqUri'], gw['oauthUri'],
-                   country, language)
 
-    def oauth_url(self):
-        return oauth_url(self.auth_base, self.country, self.language)
+        url = urljoin(self.auth_base, "spx/login/signIn")
+        query = urlencode(
+            {
+                "country": self.country,
+                "language": self.language,
+                "svc_list": SVC_CODE,
+                "client_id": CLIENT_ID,
+                "division": "ha",
+                "redirect_uri": OAUTH_REDIRECT_URI,
+                "state": uuid.uuid1().hex,
+                "show_thirdparty_login": "GGL,AMZ,FBK",
+            }
+        )
+        return "{}?{}".format(url, query)
 
-    def serialize(self) -> Dict[str, str]:
+    def dump(self):
         return {
-            'auth_base': self.auth_base,
-            'api_root': self.api_root,
-            'oauth_root': self.oauth_root,
-            'country': self.country,
-            'language': self.language,
+            "auth_base": self.auth_base,
+            "api_root": self.api_root,
+            "api2_root": self.api2_root,
+            "country": self.country,
+            "language": self.language,
         }
 
-    @classmethod
-    def deserialize(cls, data: Dict[str, Any]) -> 'Gateway':
-        return cls(data['auth_base'], data['api_root'], data['oauth_root'],
-                   data.get('country', DEFAULT_COUNTRY),
-                   data.get('language', DEFAULT_LANGUAGE))
-
 
 class Auth(object):
-    def __init__(self, gateway, access_token, refresh_token):
+    def __init__(self, gateway, oauth_url, access_token, refresh_token, user_number):
         self.gateway = gateway
+        self.oauth_url = oauth_url
         self.access_token = access_token
         self.refresh_token = refresh_token
+        self.user_number = user_number
 
     @classmethod
     def from_url(cls, gateway, url):
         """Create an authentication using an OAuth callback URL.
         """
+        oauth_url, auth_code, user_number = parse_oauth_callback(url)
+        access_token, refresh_token = login(oauth_url, auth_code)
 
-        access_token, refresh_token = parse_oauth_callback(url)
-        return cls(gateway, access_token, refresh_token)
+        return cls(gateway, oauth_url, access_token, refresh_token, user_number)
 
-    def start_session(self) -> Tuple['Session', List[Dict[str, Any]]]:
+    def start_session(self):
         """Start an API session for the logged-in user. Return the
         Session object and a list of the user's devices.
         """
-
-        session_info = login(self.gateway.api_root, self.access_token,
-                             self.gateway.country, self.gateway.language)
-        session_id = session_info['jsessionId']
-        return Session(self, session_id), get_list(session_info, 'item')
+        return Session(self)
```

Comment:
Hmm… in thinq2, is there no such thing as a "session"? Maybe we should add this to our list of things to clean up post-merge then… if there's no need to initiate sessions, there's no need for the `Session` class.